vulnerabilities/importers/istio.py

```diff
@@ -0,0 +1,193 @@
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and https://github.com/nexB/vulnerablecode/
+# The VulnerableCode software is licensed under the Apache License version 2.0.
+# Data generated with VulnerableCode require an acknowledgment.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with VulnerableCode or any VulnerableCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with VulnerableCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  VulnerableCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  VulnerableCode is a free software tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/vulnerablecode/ for support and download.
+
+import asyncio
+from typing import List, Set
+
+import yaml
+from dephell_specifier import RangeSpecifier
+from packageurl import PackageURL
+
+from vulnerabilities.data_source import Advisory
+from vulnerabilities.data_source import GitDataSource
+from vulnerabilities.data_source import Reference
+from vulnerabilities.package_managers import GitHubTagsAPI
+
+
+class IstioDataSource(GitDataSource):
+    def __enter__(self):
+        super(IstioDataSource, self).__enter__()
+
+        if not getattr(self, "_added_files", None):
+            self._added_files, self._updated_files = self.file_changes(
+                recursive=True, file_ext="md", subdir="./content/en/news/security"
+            )
+        self.version_api = GitHubTagsAPI()
+        self.set_api()
+
+    def set_api(self):
+        asyncio.run(self.version_api.load_api(["istio/istio"]))
+
+    def updated_advisories(self) -> Set[Advisory]:
+        files = self._updated_files
+        advisories = []
+        for f in files:
+            processed_data = self.process_file(f)
+            if processed_data:
+                advisories.extend(processed_data)
+        return self.batch_advisories(advisories)
+
+    def get_versions_for_pkg_from_range_list(self, version_range_list):
+        """Takes a list of version ranges(affected) of a package
+        as parameter and returns a tuple of safe package versions and
+        vulnerable package versions"""
+        all_version = self.version_api.get("istio/istio")
+        if not version_range_list:
+            return all_version, []
```

Comment:
Could you point me to an advisory where `if not version_range_list` is True ?

Reply:
I did not find such case .I added this just to be sure that version_range_list is not none.  Also while going through the advisories I have found that releases also contain a case like "All versions prior to ..." which I missed .I would make change for this ASAP.

Reply:
If you insist I can remove <code>if not version_range_list</code> case.

Reply:
> If you insist I can remove if not version_range_list case.

no :) , I was just curious.